physical/inmem.go

```diff
@@ -27,6 +29,10 @@ func NewInmem() *InmemBackend {
 func (i *InmemBackend) Put(entry *Entry) error {
 	i.l.Lock()
 	defer i.l.Unlock()
+
+	i.permitPool.Acquire()
```

Comment:
My thinking was that it only mattered that it was consistent, but I don't mind changing it.